c/tskit/tables.h

```diff
@@ -688,6 +688,40 @@ typedef struct _tsk_table_sorter_t {
 #define TSK_UNION_NO_CHECK_SHARED (1 << 0)
 #define TSK_UNION_NO_ADD_POP (1 << 1)
 
+// IBD things
+
+typedef struct _tsk_segment_t {
+    double left;
+    double right;
+    struct _tsk_segment_t *next;
+    tsk_id_t node;
+} tsk_segment_t;
+
+// Define a struct called ibd_finder_t.
+
+typedef struct {
+    tsk_id_t *pairs;
+    size_t num_pairs;
+    tsk_id_t *samples;
```

Comment:
We're not using the ``samples`` and ``num_samples`` any more are we? We should remove any unused values from the struct.

Reply:
oh, I was keeping them because they are needed to calculate pair indices the old-fashioned efficient way when you are considering all sample pairs. I can take them out for now though.